### ISSUE TITLE
[BUG] v1.1.1 upgrade may stuck in another operation (install/upgrade/rollback) by fleet

### ISSUE THREAD

**Issue by lanfon72** [harvester/harvester#3675]
### Describe the bug
Perform 3 nodes upgrade, but it stuck in _Upgrading System Service_ for hours

### To Reproduce
1. Install Harvester with 3 nodes
2. use PR https://github.com/harvester/tests/pull/690 to perform update

### Expected behavior
Upgrade should success.

## Environment:
 - Harvester ISO version: **v1.1.1** upgrade to **v1.1.2-rc3**
 - **ui-source** Option: **Auto**
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): **qemu/KVM 3 nodes**

### Support bundle
[supportbundle_a2eda14c-4899-4482-b909-65f1e91a0738_2023-03-17T14-54-51Z.zip](https://github.com/harvester/harvester/files/11002643/supportbundle_a2eda14c-4899-4482-b909-65f1e91a0738_2023-03-17T14-54-51Z.zip)

**Comment by lanfon72** [comment]
```bash
harvester-node-0:/home/rancher # kubectl logs hvst-upgrade-2jk96-apply-manifests-2tnk7 -nharvester-system | head -150
harvester: v1.1.2-rc3
harvesterChart: 1.1.2-rc3
os: Harvester v1.1.2-rc3
kubernetes: v1.24.11-rc1+rke2r1
rancher: v2.6.11
monitoringChart: 100.1.0+up19.0.3
loggingChart: 100.1.3+up3.17.7
kubevirt: 0.54.0-150400.3.7.1
minUpgradableVersion: 'v1.1.0'
rancherDependencies:
  fleet:
    chart: 100.2.3+up0.5.3
    app: 0.5.3
  fleet-crd:
    chart: 100.2.3+up0.5.3
    app: 0.5.3
  rancher-webhook:
    chart: 1.0.8+up0.2.9
    app: 0.2.9
Downloading the file from "http://upgrade-repo-hvst-upgrade-2jk96.harvester-system/harvester-iso/harvester-release.yaml" to "/tmp/tmp.Y0tViPrpQd.yaml"...
Downloading the file from "http://upgrade-repo-hvst-upgrade-2jk96.harvester-system/harvester-iso/bundle/metadata.yaml" to "/tmp/tmp.c7EW1Xsni9.yaml"...
Current version: 1.1.1
Minimum upgradable version: 1.1.0
Current version is supported.
managedchart.management.cattle.io/harvester patched
managedchart.management.cattle.io/harvester-crd patched
managedchart.management.cattle.io/rancher-monitoring patched
managedchart.management.cattle.io/rancher-monitoring-crd patched
Upgrading Rancher
Downloading the file from "http://upgrade-repo-hvst-upgrade-2jk96.harvester-system/harvester-iso/bundle//rancherd/images/rancherd-bootstrap-images-v1.1.2-rc3.txt" to "/tmp/upgrade/images/rancherd-bootstrap-images-v1.1.2-rc3.txt"...
Downloading the file from "http://upgrade-repo-hvst-upgrade-2jk96.harvester-system/harvester-iso/bundle//rancherd/images/rancherd-bootstrap-images-v1.1.2-rc3.tar.zst" to "/tmp/upgrade/images/rancherd-bootstrap-images-v1.1.2-rc3.tar.zst"...
time="2023-03-17T11:00:57Z" level=info msg="Extract mapping / => /tmp/upgrade/rancher"
time="2023-03-17T11:00:57Z" level=info msg="Checking local image archives in /tmp/upgrade/images for index.docker.io/rancher/system-agent-installer-rancher:v2.6.11"
time="2023-03-17T11:00:57Z" level=info msg="Extracting file run.sh to /tmp/upgrade/rancher/run.sh"
time="2023-03-17T11:00:57Z" level=info msg="Extracting file rancher-2.6.11.tgz to /tmp/upgrade/rancher/rancher-2.6.11.tgz"
time="2023-03-17T11:00:57Z" level=info msg="Extracting file helm to /tmp/upgrade/rancher/helm"
Rancher values:
antiAffinity: required
bootstrapPassword: admin
features: multi-cluster-management=false,multi-cluster-management-agent=false
hostPort: 0
ingress:
  enabled: false
noDefaultAdmin: false
rancherImage: rancher/rancher
rancherImagePullPolicy: IfNotPresent
rancherImageTag: v2.6.9
replicas: -3
systemDefaultRegistry: ""
tls: external
useBundledSystemChart: true
CAPI cluster fleet-local/local is provisioned (current generation: 90).
clusterrepo.catalog.cattle.io "rancher-charts" deleted
clusterrepo.catalog.cattle.io "rancher-rke2-charts" deleted
clusterrepo.catalog.cattle.io "rancher-partner-charts" deleted
setting.management.cattle.io "chart-default-branch" deleted
Release "rancher" has been upgraded. Happy Helming!
NAME: rancher
LAST DEPLOYED: Fri Mar 17 11:01:01 2023
NAMESPACE: cattle-system
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Rancher Server has been installed.

NOTE: Rancher may take several minutes to fully initialize. Please standby while Certificates are being issued, Containers are started and the Ingress rule comes up.

Check out our docs at https://rancher.com/docs/

If you provided your own bootstrap password during installation, browse to https:// to get started.

If this is the first time you installed Rancher, get started by running this command and clicking the URL it generates:

```
echo https:///dashboard/?setup=$(kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{.data.bootstrapPassword|base64decode}}')
```

To get just the bootstrap password on its own, run:

```
kubectl get secret --namespace cattle-system bootstrap-secret -o go-template='{{.data.bootstrapPassword|base64decode}}{{ "\n" }}'
```


Happy Containering!
Wait for Rancher dependencies helm releases...
Wait for Rancher dependencies rollout...
deployment "fleet-agent" successfully rolled out
deployment "fleet-controller" successfully rolled out
deployment "rancher-webhook" successfully rolled out
Wait for cluster settling down...
CAPI cluster fleet-local/local is provisioned (current generation: 102).
Upgrading Harvester Cluster Repo
deployment.apps/harvester-cluster-repo patched
Waiting for deployment "harvester-cluster-repo" rollout to finish: 1 old replicas are pending termination...
Waiting for deployment "harvester-cluster-repo" rollout to finish: 1 old replicas are pending termination...
deployment "harvester-cluster-repo" successfully rolled out
clusterrepo.catalog.cattle.io/harvester-charts patched
Upgrading Harvester
managedchart.management.cattle.io/harvester-crd patched
managedchart.management.cattle.io/harvester configured
managedchart.management.cattle.io/harvester patched
managedchart.management.cattle.io/harvester-crd patched
Waiting for ManagedChart fleet-local/harvester from generation 2
Target version: 1.1.2-rc3, Target state: ready
Current version: 1.1.2-rc3, Current state: OutOfSync, Current generation: 2
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: OutOfSync, Current generation: 2
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: OutOfSync, Current generation: 2
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: OutOfSync, Current generation: 2
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: WaitApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: WaitApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
Current version: 1.1.2-rc3, Current state: ErrApplied, Current generation: 4
Sleep for 5 seconds to retry
harvester-node-0:/home/rancher # 
```

**Comment by bk201** [comment]
hitting: https://github.com/harvester/harvester/issues/3616
```
'ErrApplied(1) [Cluster fleet-local/local: another operation (install/upgrade/rollback)
        is in progress]; storageclass.storage.k8s.io harvester-longhorn modified {"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
```

**Comment by bk201** [comment]
To add a known issue and a workaround in doc.

**Comment by lanfon72** [comment]
https://github.com/harvester/harvester/issues/3234 same topic but different case

**Comment by bk201** [comment]
To track upstream issue: https://github.com/rancher/fleet/issues/637

**Comment by lanfon72** [comment]
add the screenshot for another case (error message)
![image](https://user-images.githubusercontent.com/5169694/234836895-4ce48710-838e-49fe-a551-78792d72ebf9.png)

**Comment by cybaware** [comment]
Describe the bug
Perform 3 nodes upgrade, but it stuck in Upgrading System Service for hours
--> same issue as mentioned above

*******************************************************************************************
kubectl logs jobs/hvst-upgrade-j7bj8-apply-manifests -n harvester-system
harvester: v1.1.2
harvesterChart: 1.1.2
os: Harvester v1.1.2
kubernetes: v1.24.11+rke2r1
rancher: v2.6.11
monitoringChart: 100.1.0+up19.0.3
loggingChart: 100.1.3+up3.17.7
kubevirt: 0.54.0-150400.3.10.4
minUpgradableVersion: 'v1.1.0'
rancherDependencies:
  fleet:
    chart: 100.2.3+up0.5.3
    app: 0.5.3
  fleet-crd:
    chart: 100.2.3+up0.5.3
    app: 0.5.3
  rancher-webhook:
    chart: 1.0.8+up0.2.9
    app: 0.2.9
Downloading the file from "http://upgrade-repo-hvst-upgrade-j7bj8.harvester-system/harvester-iso/harvester-release.yaml" to "/tmp/tmp.NDLFKOn6EY.yaml"...
Downloading the file from "http://upgrade-repo-hvst-upgrade-j7bj8.harvester-system/harvester-iso/bundle/metadata.yaml" to "/tmp/tmp.nzOqf3weCb.yaml"...
Current version: 1.1.1
Minimum upgradable version: 1.1.0
Current version is supported.
managedchart.management.cattle.io/harvester patched
managedchart.management.cattle.io/harvester-crd patched
managedchart.management.cattle.io/rancher-monitoring patched (no change)
managedchart.management.cattle.io/rancher-monitoring-crd patched (no change)
Upgrading Rancher
Downloading the file from "http://upgrade-repo-hvst-upgrade-j7bj8.harvester-system/harvester-iso/bundle//rancherd/images/rancherd-bootstrap-images-v1.1.2.txt" to "/tmp/upgrade/images/rancherd-bootstrap-images-v1.1.2.txt"...
Downloading the file from "http://upgrade-repo-hvst-upgrade-j7bj8.harvester-system/harvester-iso/bundle//rancherd/images/rancherd-bootstrap-images-v1.1.2.tar.zst" to "/tmp/upgrade/images/rancherd-bootstrap-images-v1.1.2.tar.zst"...
time="2023-05-02T16:51:43Z" level=info msg="Extract mapping / => /tmp/upgrade/rancher"
time="2023-05-02T16:51:43Z" level=info msg="Checking local image archives in /tmp/upgrade/images for index.docker.io/rancher/system-agent-installer-rancher:v2.6.11"
time="2023-05-02T16:51:43Z" level=info msg="Extracting file run.sh to /tmp/upgrade/rancher/run.sh"
time="2023-05-02T16:51:44Z" level=info msg="Extracting file rancher-2.6.11.tgz to /tmp/upgrade/rancher/rancher-2.6.11.tgz"
time="2023-05-02T16:51:44Z" level=info msg="Extracting file helm to /tmp/upgrade/rancher/helm"
Rancher values:
antiAffinity: required
bootstrapPassword: admin
features: multi-cluster-management=false,multi-cluster-management-agent=false
hostPort: 0
ingress:
  enabled: false
noDefaultAdmin: false
rancherImage: rancher/rancher
rancherImagePullPolicy: IfNotPresent
rancherImageTag: v2.6.11
replicas: -3
systemDefaultRegistry: ""
tls: external
useBundledSystemChart: true
Skip update Rancher. The version is already v2.6.11
Upgrading Harvester Cluster Repo
deployment.apps/harvester-cluster-repo patched (no change)
deployment "harvester-cluster-repo" successfully rolled out
clusterrepo.catalog.cattle.io/harvester-charts patched
Upgrading Harvester
managedchart.management.cattle.io/harvester-crd patched (no change)
+ CHART_NAME=harvester
+ CHART_MANIFEST=harvester.yaml
+ case $CHART_NAME in
+ patch_harvester_ignore_default_sc
+ yq e '.spec.diff.comparePatches += [{"apiVersion": "storage.k8s.io/v1", "kind": "StorageClass", "name": "harvester-longhorn", "jsonPointers":["/metadata/annotations"]}]' harvester.yaml -i
managedchart.management.cattle.io/harvester configured
managedchart.management.cattle.io/harvester patched
managedchart.management.cattle.io/harvester-crd patched
Waiting for ManagedChart fleet-local/harvester from generation 5
Target version: 1.1.2, Target state: ready
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry
Current version: 1.1.2, Current state: ErrApplied, Current generation: 7
Sleep for 5 seconds to retry

*************************************************************************************************

To Reproduce
Steps to reproduce the behavior:

Install Harvester with 3 nodes

Expected behavior
Upgrade should success.

Environment:
Harvester ISO version: v1.1.1 upgrade to v1.1.2

Underlying Infrastructure  Baremetal 4 nodes

Additional context

***********************************************************************************************
kubectl get bundles -A
NAMESPACE     NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
fleet-local   fleet-agent-local                             1/1
fleet-local   local-managed-system-agent                    1/1
fleet-local   mcc-harvester                                 0/1                       ErrApplied(1) [Cluster fleet-local/local: another operation (install/upgrade/rollback) is in progress]
fleet-local   mcc-harvester-crd                             0/1                       ErrApplied(1) [Cluster fleet-local/local: another operation (install/upgrade/rollback) is in progress]
fleet-local   mcc-local-managed-system-upgrade-controller   1/1
fleet-local   mcc-rancher-logging                           1/1
fleet-local   mcc-rancher-logging-crd                       1/1
fleet-local   mcc-rancher-monitoring                        1/1
fleet-local   mcc-rancher-monitoring-crd                    1/1
**************************************************************************************************
Add any other context about the problem here.

-
Support bundle
-

i hope it is related to this open issue.
Thanks

**Comment by oivindoh** [comment]
I have yet to have a single upgrade go through from 1.1.1 and up across multiple attempts on all 1.1.2 rcs and now 1.1.2 release in a three-node cluster.

> https://github.com/harvester/harvester/issues/3675#issuecomment-1525418903

Hitting this exact case here with 1.1.1 to 1.1.2 upgrade on a fresh 1.1.1 cluster.

All helm charts are currently deployed, while the mcc-harvester-crd bundle is struggling: `time="2023-05-03T13:56:36Z" level=error msg="error syncing 'cluster-fleet-local-local-1a3d67d0a899/mcc-harvester-crd': handler bundle-deploy: another operation (install/upgrade/rollback) is in progress, requeuing"`:

```
❯ helm ls -A
NAME                                       	NAMESPACE                	REVISION	UPDATED                                	STATUS  	CHART                                                                                            	APP VERSION
fleet                                      	cattle-fleet-system      	2       	2023-05-03 11:38:48.385718051 +0000 UTC	deployed	fleet-100.2.3+up0.5.3                                                                            	0.5.3
fleet-agent-local                          	cattle-fleet-local-system	60      	2023-05-03 13:56:26.282171449 +0000 UTC	deployed	fleet-agent-local-v0.0.0+s-dd73d018b0bf64ec25d523466c5cd267ef6a5cfffec4d6faff1cbed6951b2
fleet-crd                                  	cattle-fleet-system      	2       	2023-05-03 11:38:58.327191026 +0000 UTC	deployed	fleet-crd-100.2.3+up0.5.3                                                                        	0.5.3
harvester                                  	harvester-system         	33      	2023-05-03 11:48:42.312331851 +0000 UTC	deployed	harvester-1.1.2                                                                                  	v1.1.2
local-managed-system-agent                 	cattle-system            	56      	2023-05-03 11:48:41.643364784 +0000 UTC	deployed	local-managed-system-agent-v0.0.0+s-48f40b284899fbe7e2014e76eaf81b4735b959f4046e035c223f86947e7a1
mcc-local-managed-system-upgrade-controller	cattle-system            	56      	2023-05-03 11:48:43.105967308 +0000 UTC	deployed	system-upgrade-controller-100.0.3+up0.3.3                                                        	v0.9.1
rancher                                    	cattle-system            	2       	2023-05-03 11:36:39.869818293 +0000 UTC	deployed	rancher-2.6.11                                                                                   	v2.6.11
rancher-logging                            	cattle-logging-system    	38      	2023-05-03 11:48:43.978686951 +0000 UTC	deployed	rancher-logging-100.1.3+up3.17.7                                                                 	3.17.7
rancher-logging-crd                        	cattle-logging-system    	50      	2023-05-03 11:48:57.176586364 +0000 UTC	deployed	rancher-logging-crd-100.1.3+up3.17.7
rancher-monitoring                         	cattle-monitoring-system 	27      	2023-05-03 09:47:24.497062798 +0000 UTC	deployed	rancher-monitoring-100.1.0+up19.0.3                                                              	0.50.0
rancher-monitoring-crd                     	cattle-monitoring-system 	39      	2023-05-03 09:47:35.976267232 +0000 UTC	deployed	rancher-monitoring-crd-100.1.0+up19.0.3
rancher-webhook                            	cattle-system            	2       	2023-05-03 11:39:06.615302996 +0000 UTC	deployed	rancher-webhook-1.0.8+up0.2.9                                                                    	0.2.9
rke2-canal                                 	kube-system              	1       	2023-04-27 13:43:47.558719955 +0000 UTC	deployed	rke2-canal-v3.24.1-build2022101103                                                               	v3.24.1
rke2-coredns                               	kube-system              	1       	2023-04-27 13:43:47.528407083 +0000 UTC	deployed	rke2-coredns-1.19.401                                                                            	1.9.3
rke2-ingress-nginx                         	kube-system              	4       	2023-04-27 13:49:25.900148476 +0000 UTC	deployed	rke2-ingress-nginx-4.1.005                                                                       	1.2.0
rke2-metrics-server                        	kube-system              	1       	2023-04-27 13:44:02.898288516 +0000 UTC	deployed	rke2-metrics-server-2.11.100-build2022101106                                                     	0.6.1
rke2-multus                                	kube-system              	1       	2023-04-27 13:43:47.549491758 +0000 UTC	deployed	rke2-multus-v3.8-build2022101103                                                                 	3.8
```

```
❯ k get bundle -A
NAMESPACE     NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
fleet-local   fleet-agent-local                             1/1
fleet-local   local-managed-system-agent                    1/1
fleet-local   mcc-harvester                                 1/1
fleet-local   mcc-harvester-crd                             0/1                       ErrApplied(1) [Cluster fleet-local/local: another operation (install/upgrade/rollback) is in progress]
fleet-local   mcc-local-managed-system-upgrade-controller   1/1
fleet-local   mcc-rancher-logging                           1/1
fleet-local   mcc-rancher-logging-crd                       1/1
fleet-local   mcc-rancher-monitoring                        0/1                       Modified(1) [Cluster fleet-local/local]; alertmanager.monitoring.coreos.com cattle-monitoring-system/rancher-monitoring-alertmanager modified {"spec":{"listenLocal":false,"paused":false,"resources":{"limits":{"cpu":"1000m"}}}}; prometheus.monitoring.coreos.com cattle-monitoring-system/rancher-monitoring-prometheus modified {"spec":{"enableAdminAPI":false,"listenLocal":false,"paused":false,"resources":{"limits":{"cpu":"1000m"}}}}
fleet-local   mcc-rancher-monitoring-crd                    1/1
```

I have not touched the monitoring setup or chart/bundle in any way.

Tried applying fleet agent from https://github.com/rancher/fleet/issues/637#issuecomment-1475927525 - but did not expect it to help since there are no relevant helm charts to roll back in this case.

FWIW these helm charts seem to do an excessive amount of deploys for no apparent reason.

**Comment by bk201** [comment]
@kinchlo @oivindoh Please follow the [workaround](https://github.com/harvester/harvester/issues/3616#issuecomment-1489892688) to rollback the stuck charts with correct chart names and namespaces. (I've revised the workaround).

@w13915984028 Do you have any insight about the modified state of mcc-rancher-monitoring bundle?

**Comment by w13915984028** [comment]
@bk201 Not yet. It is weired to encounter this `unchanged` resource marked as `modified`. Last time in @lanfon72 's ENV, no matter how we tried to edit the value back/forth, the `fleet` still complains `modified`. 

Will try to reproduce and dive into the source code.

@lanfon72 Did you re-encounter the `modified state of mcc-rancher-monitoring bundle` after that?

**Comment by bk201** [comment]
A temporary workaround could be to add the field `spec.diff.comparePatches` as we do in the harvester chart.

**Comment by oivindoh** [comment]
> @kinchlo @oivindoh Please follow the [workaround](https://github.com/harvester/harvester/issues/3616#issuecomment-1489892688) to rollback the stuck charts with correct chart names and namespaces. (I've revised the workaround).
> 
> @w13915984028 Do you have any insight about the modified state of mcc-rancher-monitoring bundle?

Ah, my apologies - I completely missed that the harvester-crd chart was in pending (just glanced over the helm ls -> deployed list).

As for monitoring chart modified being the next blocker to finish the upgrade, I ran manual `kubectl edit` on alertmanager and prometheus to add the spec properties the bundle was complaining about, and changed its cpu limit from `1` to `1000m`.

**Comment by w13915984028** [comment]
> A temporary workaround could be to add the field `spec.diff.comparePatches` as we do in the harvester chart.

@guangbochen @bk201 Seems we need to add the workaround into v1.1.2/v1.1.3 asap.

**Comment by bk201** [comment]
> > A temporary workaround could be to add the field `spec.diff.comparePatches` as we do in the harvester chart.
> 
> @guangbochen @bk201 Seems we need to add the workaround into v1.1.2/v1.1.3 asap.

We'll track the diff thing at https://github.com/harvester/harvester/issues/3839.

This issue (3675) is about https://github.com/rancher/fleet/issues/637, since we already bump Rancher and fleet, we can see if it happens again.

**Comment by bk201** [comment]
The current Fleet version in both v1.2/v1.3 branches has this fix and we didn't notice it in 1.3.0 development time. Please re-open if the issue present again.
